Tests/NIOWebSocketTests/NIOWebSocketClientUpgraderTests.swift

```diff
@@ -0,0 +1,39 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import NIOWebSocket
+
+
+/// a mock random number generator which will return the given `numbers` in order
+fileprivate struct TestRandomNumberGenerator: RandomNumberGenerator {
+    var numbers: [UInt64]
+    var nextRandomNumberIndex: Int
+    init(numbers: [UInt64], nextRandomNumberIndex: Int = 0) {
+        self.numbers = numbers
+        self.nextRandomNumberIndex = nextRandomNumberIndex
+    }
+    mutating func next() -> UInt64 {
+        defer { nextRandomNumberIndex += 1 }
+        return numbers[nextRandomNumberIndex % numbers.count]
+    }
+}
+
+final class NIOWebSocketClientUpgraderTests: XCTestCase {
+    func testRandomRequestKey() {
```

Comment:
Can we add a test of the system random number generator entry point as well?

Reply:
Good point. Done ✅